selfhost/typechecker.jakt

```diff
@@ -4044,6 +4044,74 @@ struct Typechecker {
                                         ))
                                         .add_var_to_scope(scope_id: new_scope_id, name: variant_argument.binding, var_id, span)
                                     }
+                                    StructLike(name, fields) => {
+                                        covered_variants.push(name)
+
+                                        mut field_variables: [CheckedVariable] = []
+                                        for var_id in fields.iterator() {
+                                            field_variables.push(.program.get_variable(var_id))
+                                        }
+                                        mut seen_names: [String] = [] // FIXME: HashSet
```

Comment:
Any reason not to use `mut seen_names: {String} = {}`?

Reply:
Did not know about that!